perses/tests/test_relative.py

```diff
@@ -937,3 +937,167 @@ def test_flattenedHybridTopologyFactory_energies():
     # Test alanine dipeptide repartitioned htf with flattened torsions and exceptions in vacuum
     atp, system_generator = generate_atp()
     flattenedHybridTopologyFactory_energies(atp.topology, '1', atp.system, atp.positions, system_generator, repartitioned=True)
+
+
+def test_RESTCapableHybridTopologyFactory_energies():
+    """
+    Test whether the difference in the nonalchemical zero and alchemical zero states is the forward valence energy.  Also test for the one states.
+
+    By default, runs point energy (of the positions in the htf) test and skips the MD test.
+
+    """
+    import tempfile
+    import pickle
+
+    from perses.tests.test_topology_proposal import generate_atp, generate_dipeptide_top_pos_sys
+    from perses.app.relative_point_mutation_setup import PointMutationExecutor
+    from perses.tests.utils import validate_endstate_energies_point, validate_endstate_energies_md
+
+    # Alanine dipeptide in vacuum
+    print("Alanine dipeptide in vacuum")
+    atp, system_generator = generate_atp(phase='vacuum')
+    htf = generate_dipeptide_top_pos_sys(atp.topology,
+                                         'THR',
+                                         atp.system,
+                                         atp.positions,
+                                         system_generator,
+                                         conduct_htf_prop=True,
+                                         generate_rest_capable_hybrid_topology_factory=True,
+                                         validate_endstate_energy=False)
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        with open(os.path.join(temp_dir, "atp_vacuum.pickle"), "wb") as f:
+            pickle.dump(htf, f)
+
+        # Test at lambda = 0
+        with open(os.path.join(temp_dir, "atp_vacuum.pickle"), "rb") as f:
+            htf = pickle.load(f)
+        validate_endstate_energies_point(htf, minimize=True)
+
+        with open(os.path.join(temp_dir, "atp_vacuum.pickle"), "rb") as f:
+            htf = pickle.load(f)
+        validate_endstate_energies_md(htf, n_steps=750) # Run 3 ps of MD
```

Comment:
@mikemhenry @ijpulidos : For each of my test cases in `test_RESTCapableHybridTopologyFactory_energies()`, I generate a htf for the test case, save it in a temporary directory, and then call `validate_endstate_energies_point()` and  `validate_endstate_energies_md()` for each endstate. `validate_endstate_energies_md()` involves running `n_steps` of MD.. is there an easy way to specify that `validate_endstate_energies_md()`  calls should be run on GPUs (or maybe ignored for GHA), but also keep the calls in `test_RESTCapableHybridTopologyFactory_energies()`? Otherwise, these will slow down our GHA job runs a lot. I want to keep the `validate_endstate_energies_md()` calls where they are now because ideally we would only generate each test case htf once.

Reply:
Hmm, good question. I guess there are a bunch of options that we can explore.

 * Do we need to test every case or system? I see we test alanine, peptide and barstar. Are these testing different parts of the code or one could say that doing one is enough to test code functionality? In that case we would think about just running one (the fastest) in GHA.
 * Another ting we might be able to do is to separate the tests in different ones making it more modular, initially we could think about separating them by systems (one for alanine, another for peptide and another for barstar). But we might also try separating each of these further, i.e. one for `validate_endstate_energies_point` and another for `validate_endstate_energies_md` or similar. And all the possible combinations options you can imagine, the goal is to make it modular.
 
Summarizing, big questions here probably are, do we need all these tests to be run in the same module (at the same time)? Do we have a minimal working example that hopefully runs fast with GHA that test the code functionality? If that's the case, we might just use that for every PR, and let the other ones (if we split them) to be run only when the GPU CI takes action.

Reply:
>Do we need to test every case or system? I see we test alanine, peptide and barstar. Are these testing different parts of the code or one could say that doing one is enough to test code functionality?

The idea was to provide test cases for simple systems as well as more complicated systems. Alanine dipeptide in vacuum is the simplest -- it doesn't have any environment atoms (protein or solvent) and doesn't use PME. Alanine dipeptide in solvent increases in complexity by a little bit -- it has environment atoms (solvent, but no protein) and uses PME. The 8mer peptide has both types of environment atoms and also uses PME. Barnase:barstar has the same features as the 8mer, but is a system we would actually be interested in running. In theory, we could probably get away with only testing barnase:barstar, as its the most complicated test case, but having the simpler test cases helps us isolate where potential problems may be coming from.

>But we might also try separating each of these further, i.e. one for validate_endstate_energies_point and another for validate_endstate_energies_md or similar

I don't think we want to do this, because separating them will require generating the same htf two different times (assuming we are saving them in a temporary directory). This is the constraint I pointed out in my initial question.

> do we need all these tests to be run in the same module (at the same time)?

yes, i think so (for the reasons I described in my response above this one), unless we can somehow avoid using temporary files. 

I think we should discuss this at the next perses dev sync. But the crux of my question is whether we can keep all the test calls in one function and still specify which ones should be only run on GPUs... I'm guessing the answer is no.

Reply:
Ah got it! And yes, as far as I know you can only mark tests at a function level (marking is what we are using to run on GPU). But it should still be possible to split the test in systems, and think if testing the simplest case(s) is "enough" for our CI requirements.